addon/components/file-picker.js

```diff
@@ -73,13 +73,31 @@ export default Component.extend({
     }
   },
 
+  _invalidExtension: function(files) {
+    let accept = this.get('accept');
+
+    if (accept === '*') {
+      return;
+    }
+
+    let validExtensions = accept.split(',');
+
+    let fileExtensions = files.map(file => `.${file.filename.split('.').slice(-1)[0]}`);
+
+    return fileExtensions.some(extension => validExtensions.indexOf(extension) === -1);
+  },
+
   handleFiles: function(files) {
     if (typeof(this.filesAreValid) === 'function') {
```

Comment:
Can you move the block into a function `_validate` and call your function in `_validate`?